src/lib/mina_base/snapp_predicate.ml

```diff
@@ -1052,12 +1047,19 @@ module Account_type = struct
     | _ ->
         assert false
 
-  let to_input x = Random_oracle_input.bitstring (to_bits x)
+  let to_input x =
+    let open Random_oracle_input.Chunked in
+    Array.reduce_exn ~f:append
```

Comment:
I wonder if it would be possible to make a helper to encapsulate this sort of thing so it doesn't need to be repeated in many places.

Reply:
Now that nearly everything has a `to_input`, it would probably be even cleaner to replace most of these with a `deriving random_oracle_input`, and then we definitely won't miss any fields or repeat ourselves.